language/stdlib/modules/libra_configs.move

```diff
@@ -31,38 +38,113 @@ module LibraConfig {
+    // reconfiguration.
+    public fun set<Config: copyable>(payload: Config) acquires T, Configuration {
+        let addr = default_config_address();
+        Transaction::assert(::exists<T<Config>>(addr), 24);
+        Transaction::assert(
+            ::exists<ModifyConfigCapability<Config>>(Transaction::sender())
+             || Transaction::sender() == Association::root_address(),
+            24
+        );
+
+        let config = borrow_global_mut<T<Config>>(addr);
+        config.payload = payload;
+
+        reconfigure_();
+    }
 
+    // Set a config item to a new value and trigger a reconfiguration.
+    public fun set_with_capability<Config: copyable>(
+        _cap: &ModifyConfigCapability<Config>,
+        payload: Config
+    ) acquires T, Configuration {
+        let addr = default_config_address();
         Transaction::assert(::exists<T<Config>>(addr), 24);
         let config = borrow_global_mut<T<Config>>(addr);
         config.payload = payload;
 
         reconfigure_();
     }
 
-    // Publish a new config item to a new value and trigger a reconfiguration.
-    public fun publish_new_config<Config: copyable>(payload: Config) {
-        // TODO: impose proper permission checks
-        // Callable by the any address for now.
+    // Publish a new config item. The caller will use the returned ModifyConfigCapability to specify the access control
+    // policy for who can modify the config.
+    public fun publish_new_config_with_capability<Config: copyable>(payload: Config): ModifyConfigCapability<Config> {
+        Transaction::assert(
+            Association::has_privilege<CreateConfigCapability>(Transaction::sender()),
+            1
+        );
 
         move_to_sender(T{ payload });
         // We don't trigger reconfiguration here, instead we'll wait for all validators update the binary
         // to register this config into ON_CHAIN_CONFIG_REGISTRY then send another transaction to change
         // the value which triggers the reconfiguration.
+
+        return ModifyConfigCapability<Config> {}
+    }
+
+    // Publish a new config item. Only the config address can modify such config.
+    public fun publish_new_config<Config: copyable>(
+        payload: Config,
+    ) {
+        Transaction::assert(
+            Association::has_privilege<CreateConfigCapability>(Transaction::sender()),
+            1
+        );
+
+        move_to_sender(ModifyConfigCapability<Config> {});
+        move_to_sender(T{ payload });
+        // We don't trigger reconfiguration here, instead we'll wait for all validators update the binary
+        // to register this config into ON_CHAIN_CONFIG_REGISTRY then send another transaction to change
+        // the value which triggers the reconfiguration.
+    }
+
+    // Publish a new config item. Only the delegated address can modify such config after redeeming the capability.
+    public fun publish_new_config_with_delegate<Config: copyable>(
```

Comment:
Is there a test that exercises this + `claim_delegated_modify_config`?

Reply:
Just added the test!